src/Microsoft.AspNet.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -38,6 +38,34 @@ public Task<int> ReadAsync(ArraySegment<byte> buffer, CancellationToken cancella
             return result;
         }
 
+        public async Task Consume(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Task<int> result;
+            var firstLoop = true;
+            do
+            {
+                result = ReadAsyncImplementation(default(ArraySegment<byte>), cancellationToken);
```

Comment:
Was copying the function above. Following the logic of that function; the task hasn't completed because no data; check whether their is no data because the client is waiting for a "100 continue" else the await will block infinitely as it waiting for data that's dependent on a send; if so do the send and return the incomplete task?

Since we are doing the full loop here; if we pass over that condition; we know the task has already completed. If the result is `0` then return as everything is done; otherwise move to next loop; so we only await if the task is not complete.